src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -275,6 +278,60 @@ export class FilesRenderer implements ICompressibleTreeRenderer<ExplorerItem, Fu
 	private _onDidChangeActiveDescendant = new EventMultiplexer<void>();
 	readonly onDidChangeActiveDescendant = this._onDidChangeActiveDescendant.event;
 
+	private readonly hoverDelegate = new class implements IHoverDelegate {
+
+		readonly placement = 'element';
+
+		get delay() {
+			return this.configurationService.getValue<number>('workbench.hover.delay');
+		}
+
+		constructor(
+			private readonly configurationService: IConfigurationService,
+			private readonly hoverService: IHoverService
+		) { }
+
+		showHover(options: IHoverDelegateOptions, focus?: boolean): IHoverWidget | undefined {
+			let element: HTMLElement;
+			if (options.target instanceof HTMLElement) {
+				element = options.target;
+			} else {
+				element = options.target.targetElements[0];
+			}
+
+			const child = element.children[0] as Element | undefined;
+			const childOfChild = child?.children[0] as HTMLElement | undefined;
+			let overflowed = false;
+			if (childOfChild && child) {
+				const width = child.clientWidth;
+				const childWidth = childOfChild.offsetWidth;
+				// Check if element is overflowing its parent container
+				overflowed = width <= childWidth;
+			}
+
+			const hasDecoration = element.classList.toString().includes('monaco-decoration-iconBadge');
+			// If it's overflowing or has a decoration show the tooltip
+			overflowed = overflowed || hasDecoration;
+			// TODO @lramos15 find a better way to do this, grabs the indent element to position hover above the label
+			const indentGuideElement = element.parentElement?.parentElement?.children[0] as HTMLElement | undefined;
+
+			if (!indentGuideElement) {
+				return;
+			}
+
+			return overflowed ? this.hoverService.showHover({
+				...options,
+				target: indentGuideElement,
+				compact: true,
+				additionalClasses: ['explorer-item-hover'],
```

Comment:
The icon would be a nice addition and I see you said we can defer that, but I think aligning the position and font exactly with underneath would be really good to go out with this release. It's one of the things that makes the Windows Explorer feature so nice imo.

<img width="835" alt="image" src="https://user-images.githubusercontent.com/2193314/218790382-27efff95-5f5b-4fab-a8fd-664459c21342.png">

Reply:
<img width="271" alt="image" src="https://user-images.githubusercontent.com/4544166/218829776-07af8f4b-f4e9-4f86-b668-c6673af6a45e.png">
Hopefully this is close eough

Reply:
Looks better 👍 

You could maybe align it easy via `getComputedStyle` and then applying all the relevant font values to the hover element?